sycl/include/sycl/ext/intel/esimd/detail/math_intrin.hpp

```diff
@@ -32,45 +32,68 @@
 template <typename T> extern __DPCPP_SYCL_EXTERNAL T __spirv_ocl_native_exp2(T);
 template <typename T, int N>
 extern __DPCPP_SYCL_EXTERNAL __ESIMD_raw_vec_t(T, N)
-    __spirv_ocl_native_exp2(__ESIMD_raw_vec_t(T, N));
+    __spirv_ocl_native_exp2(__ESIMD_raw_vec_t(T, N)) __ESIMD_INTRIN_END;
 
 template <typename T>
 extern __DPCPP_SYCL_EXTERNAL T __spirv_ocl_native_recip(T);
 template <typename T, int N>
 extern __DPCPP_SYCL_EXTERNAL __ESIMD_raw_vec_t(T, N)
-    __spirv_ocl_native_recip(__ESIMD_raw_vec_t(T, N));
+    __spirv_ocl_native_recip(__ESIMD_raw_vec_t(T, N)) __ESIMD_INTRIN_END;
 
 template <typename T> extern __DPCPP_SYCL_EXTERNAL T __spirv_ocl_native_cos(T);
 template <typename T, int N>
 extern __DPCPP_SYCL_EXTERNAL __ESIMD_raw_vec_t(T, N)
-    __spirv_ocl_native_cos(__ESIMD_raw_vec_t(T, N));
+    __spirv_ocl_native_cos(__ESIMD_raw_vec_t(T, N)) __ESIMD_INTRIN_END;
 
 template <typename T> extern __DPCPP_SYCL_EXTERNAL T __spirv_ocl_native_log2(T);
 template <typename T, int N>
 extern __DPCPP_SYCL_EXTERNAL __ESIMD_raw_vec_t(T, N)
-    __spirv_ocl_native_log2(__ESIMD_raw_vec_t(T, N));
+    __spirv_ocl_native_log2(__ESIMD_raw_vec_t(T, N)) __ESIMD_INTRIN_END;
 
 template <typename T>
 extern __DPCPP_SYCL_EXTERNAL T __spirv_ocl_native_rsqrt(T);
 template <typename T, int N>
 extern __DPCPP_SYCL_EXTERNAL __ESIMD_raw_vec_t(T, N)
-    __spirv_ocl_native_rsqrt(__ESIMD_raw_vec_t(T, N));
+    __spirv_ocl_native_rsqrt(__ESIMD_raw_vec_t(T, N)) __ESIMD_INTRIN_END;
 
 template <typename T> extern __DPCPP_SYCL_EXTERNAL T __spirv_ocl_native_sin(T);
 template <typename T, int N>
 extern __DPCPP_SYCL_EXTERNAL __ESIMD_raw_vec_t(T, N)
-    __spirv_ocl_native_sin(__ESIMD_raw_vec_t(T, N));
+    __spirv_ocl_native_sin(__ESIMD_raw_vec_t(T, N)) __ESIMD_INTRIN_END;
 
 template <typename T> extern __DPCPP_SYCL_EXTERNAL T __spirv_ocl_native_sqrt(T);
 template <typename T, int N>
 extern __DPCPP_SYCL_EXTERNAL __ESIMD_raw_vec_t(T, N)
-    __spirv_ocl_native_sqrt(__ESIMD_raw_vec_t(T, N));
+    __spirv_ocl_native_sqrt(__ESIMD_raw_vec_t(T, N)) __ESIMD_INTRIN_END;
 
 template <typename T>
 extern __DPCPP_SYCL_EXTERNAL T __spirv_ocl_native_powr(T, T);
 template <typename T, int N>
 __ESIMD_INTRIN __ESIMD_raw_vec_t(T, N)
-    __spirv_ocl_native_powr(__ESIMD_raw_vec_t(T, N), __ESIMD_raw_vec_t(T, N));
+    __spirv_ocl_native_powr(__ESIMD_raw_vec_t(T, N),
+                            __ESIMD_raw_vec_t(T, N)) __ESIMD_INTRIN_END;
+
+template <typename T> extern __DPCPP_SYCL_EXTERNAL T __spirv_ocl_fabs(T);
+template <typename T, int N>
+extern __DPCPP_SYCL_EXTERNAL __ESIMD_raw_vec_t(T, N)
+    __spirv_ocl_fabs(__ESIMD_raw_vec_t(T, N)) __ESIMD_INTRIN_END;
```

Comment:
My stand haven't changed here: ideally, we shouldn't declare those functions in headers and we should use automatically declared versions provided by the compiler. The reason for that is because the latter are more efficient and we will have less code in our headers.

However, I understand that it requires a significant refactoring/rework, because ESIMD allows vectors of any lengths, whilst core SPIR-V is limited to 2, 3, 4, 8, 16. Therefore, I do not intend to block this PR.

I wonder, however, if some disclaimer/instruction should be put directly into the header so that next time it is updated, those declarations are added properly. I understand that it could be missed, but at least that's some documentation within the code and not within a commit message which needs to be retrieved through `git blame`.

Reply:
Agree with you, I would absolutely prefer to use the Clang-generated ones, but given the ESIMD requirements I can't.

I'll add that warning, thanks.